tests/sm_ut/sm_ut.c

```diff
@@ -531,6 +578,36 @@ TEST_FUNCTION(sm_close_end_switches_state_to_SM_CREATED) /*allows sm_open_begin
     sm_destroy(sm);
 }
 
+TEST_FUNCTION(sm_close_end_switches_state_to_SM_CREATED_leaves_SM_FAULTED_BIT_set)
```

Comment:
Let's add the tested specs

Reply:
Good point, added some specs about calls being allowed when faulted and this not resetting faulted

---
In reply to: [579550722](https://github.com/Azure/c-util/pull/106#discussion_r579550722) [](ancestors = 579550722)